packages/ResizeDetector/src/ResizeDetector.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable react/require-default-props */
+import React from "react";
+import useResizeDetector from "use-resize-observer/polyfilled";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import { useCallbackDebouncer } from "@paprika/helpers";
+import * as sc from "./ResizeDetector.styles";
+
+// TODO: remove this hack after @paprika/helpers is migrated to TS
+const ShirtSizes_: any = ShirtSizes;
+
+export type Dimensions = { width: number; height: number };
+export type ResizeDetectorContextValue = { width: number; height: number; breakpointSize: string };
+
+interface ResizeDetectorProps {
+  /** The width at which the size will change from the default (medium) to large. 0 or null value will disable. */
+  breakpointLarge: number;
+
+  /** The width at which the size will change from small to the default (medium). 0 or null value will disable. */
+  breakpointSmall: number;
+
+  /** Content to be wrapped which will be provided with live dimensions and (tshirt) size values.  */
+  children: React.ReactNode | ((context: ResizeDetectorContextValue) => React.ReactNode);
+
+  /** The ms delay before firing resize events / making live updates. */
+  debounceDelay: number;
+
+  /** If the container will match its parent's width like a block level element (width: 100%). */
+  isFullWidth: boolean;
+
+  /** If the container will match its parent's height (height: 100%). */
+  isFullHeight: boolean;
+
+  /** Callback that fires when the size change crosses a breakpoint threshold (returns new T-Shirt size value).  */
+  onBreak?: (breakpointSize: string) => void;
+
+  /** Callback that fires when the size changes (returns new width + height values).  */
+  onResize?: (dimensions: Dimensions) => void;
+}
+
+const MAX_WAIT = 1000;
+
+const ResizeContext = React.createContext({ width: 0, height: 0, breakpointSize: ShirtSizes_.MEDIUM });
+
+export function useResizeDetectorContext(): ResizeDetectorContextValue {
```

Comment:
How about just `useResizeDetector()`? 

Generally I prefer shorter when possible – though sometimes too short is too ambiguous (like `size` 😄)

Reply:
agree